### PR TITLE
Random Agent

### PR DESCRIPTION
**Pull request by GoldenMurrani**
Agent that walks randomly based on seed from world

### CODE REVIEW

**Review by Kanakmk — APPROVED** [review]
The Random Agent looks good!

File: tests/build/CMakeCache.txt
Comment (mdkdoc15) [inline tests/build/CMakeCache.txt:1]: Im unsure if all these files need to be here, due to the fact they are build files

File: source/core/AgentBase.hpp
Comment (mdkdoc15) [inline source/core/AgentBase.hpp:1]: do we want this inventory class? I think the prof was implementing it on his own. If that is the case then we should remove all files associated with the inventory

**Review by mdkdoc15 — CHANGES_REQUESTED** [review]
Agent looks good, there seems to be a lot of build files that should be removed, Additionally unsure if the Item Inventory need to be included

**Review by mdkdoc15 — APPROVED** [review]
Changes made looks good

**Pull request merged**